src/modelinstance.cpp

```diff
@@ -82,13 +82,19 @@ Status ModelInstance::loadInputTensors(const ModelConfig& config, const DynamicM
         if (name == ANONYMOUS_INPUT_NAME) {
             continue;
         }
-        if (networkInputs.count(name) == 0) {
+        if (networkInputs.count(name) == 1 && config.getMappingInputByKey(name) != "") {
+            SPDLOG_WARN("Config shape - {} is mapped by {}. Changes will not apply", name, config.getMappingInputByKey(name));
+        } else if (networkInputs.count(name) == 0 && networkInputs.count(config.getRealInputNameByValue(name)) == 0) {
             SPDLOG_WARN("Config shape - {} not found in network", name);
             return StatusCode::CONFIG_SHAPE_IS_NOT_IN_NETWORK;
         }
     }
     for (const auto& [name, _] : config.getLayouts()) {
-        if (networkInputs.count(name) == 0 && network->getOutputsInfo().count(name) == 0) {
+        if (networkInputs.count(name) == 1 && config.getMappingInputByKey(name) != "") {
+            SPDLOG_WARN("Config layout - {} is mapped by {}. Changes will not apply", name, config.getMappingInputByKey(name));
```

Comment:
Are you sure we do not want to return an error similarly to line 99?

Reply:
Should we then return someting like `StatusCode::CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME` and same with shape everytime we find real input/output name used in config, which is mapped in mapping_config.json? Technically we would need to create 2 new StatusCodes and after every warn written  we would need to return appropriate StatusCode.